catalog/app/containers/Bucket/PackageRevisions.js

```diff
@@ -428,21 +432,56 @@ export default function PackageRevisions({
     }
   })
 
-  const revisionCountData = useRevisionCountData({ bucket, name })
+  const [key, setKey] = React.useState(1)
+
+  const revisionCountData = useRevisionCountData({ bucket, name, key })
   const revisionsData = useRevisionsData({
     bucket,
     name,
     page: actualPage,
     perPage: PER_PAGE,
+    key,
   })
-  const countsData = useCountsData({ bucket, name })
+  const countsData = useCountsData({ bucket, name, key })
+
+  const onExited = React.useCallback(
+    (res) => {
+      // refresh data if new revision of current package was pushed
+      if (res && res.pushed && res.pushed.name === name) {
+        setKey(R.inc)
+        return true
+      }
+      return false
+    },
+    [name, setKey],
+  )
+
+  const updateDialog = usePackageUpdateDialog({ bucket, name, onExited })
```

Comment:
Why do you use `component({ prop1, prop2 })` instead of regular `<Component prop1={prop1} prop2={prop2} />`? Because of memoizing?

Reply:
i dont quite get the question. i use a hook here, bc i need to be able to access some trigger functions i.e. `open` and `close`. it's not very convenient to expose triggers when using a component (you need to use refs and stuff), so using hook here makes more sense.

Reply:
>  i need to be able to access some trigger functions i.e. open and close

I see now. It's an answer